File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> diffToIndex = new HashMap<Integer, Integer>();
        int[] result = new int[2];
        for (int i = 0; i < nums.length; i++) {
            diffToIndex.put(target - nums[i], i);
        }
        
        for (int i = 0; i < nums.length; i++) {
            if (diffToIndex.containsKey(nums[i])) {
                int current = diffToIndex.get(nums[i]);
                if (current != i) {
                    result[0] = i;
                    result[1] = current;
                }
            }
        }
        
        return result;
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> diffToIndex;
        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];
            if (diffToIndex.count(complement)) {
                return {diffToIndex[complement], i};
            }
            
            diffToIndex[nums[i]] = i;
        }
        
        return {};
    }
};